packages/astro/src/content/runtime.ts

```diff
@@ -208,6 +218,11 @@ export function createGetDataEntryById({ getEntryImport }: { getEntryImport: Get
 				collection,
 			};
 		}
+		if (!collectionNames.has(collection)) {
+			// eslint-disable-next-line no-console
+			console.warn(`The collection ${JSON.stringify(collection)} does not exist.`);
```

Comment:
Is this the right behaviour? It's not clear whether a missing collection should throw or just warn.

Reply:
It also warns in `main` now so I think warning is fine https://github.com/withastro/astro/blob/026e8baf3323e99f96530999fd32a0a9b305854d/packages/astro/src/content/runtime.ts#L64-L72

Reply:
Yeah, that's the only reason I used it. I was trying to follow the current behaviour.